File: Backend (Springboot)/Annapurna/src/main/java/com/annapurna/domain/Tiffin.java
package com.annapurna.domain;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tiffin {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String name;
	private String description;
	private double price;
	
	@Lob
	@Column(name = "image", columnDefinition = "LONGBLOB")
	private byte[] image;

	private String imageUrl;

	@ManyToOne
	@JoinColumn(name = "vendor_id")
	private User vendor; // Vendor is also a user with role=VENDOR

	@OneToMany(mappedBy = "tiffin", cascade = CascadeType.ALL)
	private List<Order> orders;

	@Enumerated(EnumType.STRING)
	private TiffinCategory category;

}

File: Backend (DNET)/Annapurna/Models/Entities/Tiffin.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Annapurna.Models.Enums;

namespace Annapurna.Models.Entities
{
    public class Tiffin
    {
        [Key]
        public long Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public double Price { get; set; }

        public byte[] Image { get; set; }

        public string ImageUrl { get; set; }

        public TiffinCategory Category { get; set; }

        // Foreign key
        public long VendorId { get; set; }

        public User Vendor { get; set; }

        public ICollection<Order> Orders { get; set; }
    }
}
